src/main/java/sevensmurfs/rehub/service/TherapyService.java

```diff
@@ -17,6 +24,37 @@ public class TherapyService {
 
     private final TherapyResultService therapyResultService;
 
+    private final PatientRepository patientRepository;
+
+    private final DoctorRepository doctorRepository;
+
+    @Transactional
+    public Therapy createTherapy(TherapyRequest newTherapy) {
+
+        log.debug("Creating therapy entity.");
+        Patient patient = patientRepository.findById(newTherapy.getPatientId()).orElseThrow(
+                () -> new IllegalArgumentException("Patient with given id does not exists."));
+
+        Doctor doctor = doctorRepository.findByFirstAndLastName(newTherapy.getDoctorFirstName(), newTherapy.getDoctorLastName());
+
+        if(doctor == null) throw new IllegalArgumentException(String.format("Doctor '%s' '%s' not found!", newTherapy.getDoctorFirstName(), newTherapy.getDoctorLastName()));

```

Comment:
This won't be needed when repository JPA method is fixed.

Reply:
With method returning Optional, I should still check if doctor is found with _If(doctor.isEmpty()){ ..._  , right?